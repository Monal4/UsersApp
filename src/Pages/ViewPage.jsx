import React, { useState, useEffect } from 'react';
import { useHistory } from 'react-router-dom';
import {connect} from 'react-redux';
import {getUsers} from '../redux-files/user-reducer/user-actions';
import Users from '../Components/users-view-component/usersView.component';

const ViewPage = ({getUsers, users}) => {

    const[ searchField, setSearchField] = useState('');

    useEffect(() => {
        getUsers()
    }, [getUsers]);

    const handleInput = (value) => {
        setSearchField(value);
    };

    const filteredUsers = users.filter((user) => (
        user.name.toLowerCase().startsWith(searchField.toLowerCase())    
    ));

    const history = useHistory();

    const handlePost = (id) => {
        history.push("/posts/"+id)
    }

    return(
        <Users users={filteredUsers} handleInput={handleInput} handlePost={handlePost}/>
    )
}

const mapPropsToState = state => {return {users:state.users.users}};
export default connect(mapPropsToState, {getUsers})(ViewPage);