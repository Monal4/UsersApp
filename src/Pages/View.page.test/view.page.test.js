import React from 'react';
import { shallow } from "enzyme";
import '../../setup';
import ViewPage from "../ViewPage";

import {Provider} from 'react-redux';
import store from '../../redux-files/store'

describe('View Page Component test', () => {
    
    it('is a snapshot for ViewPage', () => {
        expect( shallow(
            <Provider store={store}>
                <ViewPage/>
            </Provider>
            ) ).toMatchSnapshot();
    });

});