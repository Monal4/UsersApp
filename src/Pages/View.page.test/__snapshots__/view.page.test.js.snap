// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`View Page Component test is a snapshot for ViewPage 1`] = `ShallowWrapper {}`;
